### ISSUE TITLE
new rule - variable is not used before it's assigned again

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#10107]
```py
a = 1 # error: asignment is useless because it's not used before being assigned again
a = 2
print(a)
```

**Comment by MichaReiser** [comment]
I like that! This rule is similar to dead code analysis (and should maybe be integrated into a dead code) and would profit from a data flow analysis framework so that it also catches

```python
a = 1
if x:
	a = 2
else: 
	a = 3
```

**Comment by mikeleppane** [comment]
This rule would be helpful and considered part of the dead code analysis. I might start looking into this soonish.

**Comment by nijel** [comment]
Slightly related would be:

```python
a = 1
print(a)
a = 2 # error: assignment is useless because it's not acessed again
```

**Comment by DetachHead** [comment]
> This sounds like something [`F811`](https://docs.astral.sh/ruff/rules/redefined-while-unused/) should catch, but it currently doesn't.

i think that rule is for redeclarations whereas this issue is for reassignments (the difference being whether a type annotation is used). relevant conversation: https://github.com/astral-sh/ty/issues/2987#issuecomment-4017995151
